src/__tests__/fixtures/regression.css

```diff
@@ -0,0 +1,7 @@
+@custom-selector :--button .button;
+
+:--button {
+  &--active {
```

Comment:
Given:

```css
@custom-selector :--blah .blah p;

:--blah {
  &--wat { ... }
}
```

this desugars to:

```css
--wat:matches(:--blah) { ... }
```

exactly like the spec says. ^_^

You can then desugar the custom selector as well, getting:

```css
--wat:matches(.blah p) { ... }
```

As it so happens, this is guaranteed to match nothing, since no element has both "p" and "--wat" as its tagname.